cli/tests/integration_tests.rs

```diff
@@ -16,13 +16,16 @@ use tempfile::TempDir;
 fn std_tests() {
   let dir = TempDir::new().expect("tempdir fail");
   let std_path = util::root_path().join("std");
+  let std_config = std_path.join("tsconfig_test.json");
   let status = util::deno_cmd()
     .env("DENO_DIR", dir.path())
     .current_dir(std_path) // TODO(ry) change this to root_path
     .arg("test")
     .arg("--unstable")
     .arg("--seed=86") // Some tests rely on specific random numbers.
     .arg("-A")
+    .arg("--config")
+    .arg(std_config.to_str().unwrap())
```

Comment:
It's a bit unlucky that testing `std/` now requires custom config... I'm not fully grasping why, @kitsonk could you explain?

Reply:
It only requires it as part of the CI.  The explanation is a bit complex.  TypeScript with type checking allows code that is not transpile only safe (be it Babel, swc, or transpileOnly).  There are a couple options which can help ensure TypeScript code doesn't "surprise" with runtime errors when transpiling.  If we added them as standard options to Deno right now, it would be a breaking change and a lot of existing Deno code would simply not work, but at the same time, we don't want to commit more code to `std` that is transpile "unsafe" so the add of the configuration file holds the `std` code to a higher standard than Deno has by default at this point in time.

My advice would be with Deno 2, that we harden down the config so that we try to ensure type checked code "breaks" when it contains unsafe type directed emits.

The biggest things that just "break" in transpile are:

- Re-exporting type only entities with `export ... from ...` instead of `export type ... from ...`
- An import of only types with `import ... from` instead of `import type ... from`
- Use of `const enum`
- Use of legacy `export =` and `import =` TypeScript syntax.

We could remove it, but the chances are that people will try to commit non-transpile safe code to `std`, which will break and get reported back as a runtime error when people are using `--no-check`.

Reply:
Thank you for thorough explanation, it all makes sense now. 

> My advice would be with Deno 2, that we harden down the config so that we try to ensure type checked code "breaks" when it contains unsafe type directed emits.

I do agree with you on this one.